File: samples/bot-tab-conversations/csharp/Source/ConversationalTabs.Web/ClientApp/src/models/SupportDepartment.ts
import { CustomerInquiry } from "./CustomerInquiry";
import { MsTeamsBotData } from "./MsTeamsBotData";

export type SupportDepartment = {
  supportDepartmentId: string;
  title: string;
  description: string;
  teamChannelId: string;
  tenantId: string;
  proactiveBotData: MsTeamsBotData;
  subEntities: CustomerInquiry[];
};

export type SupportDepartmentInput = {
  title: string;
  description: string;
  teamChannelId: string;
  teamId: string;
  groupId: string;
  tenantId: string;
};

File: samples/bot-tab-conversations/csharp/Source/ConversationalTabs.Domain/Models/SupportDepartment.cs
// Copyright (c) Microsoft Corporation. All rights reserved.

namespace Microsoft.Teams.Samples.ConversationalTabs.Domain.Models;

using System.Text.Json.Serialization;

/// <summary>
/// Model for a Support Department, which is the Entity in our example
/// </summary>
public record SupportDepartment(
    [property: JsonPropertyName("supportDepartmentId")]
    string Id,
    string Title,
    string Description,
    string GroupId,
    string TeamChannelId,
    string TenantId,
    MsTeamsBotData ProactiveBotData,
    ICollection<CustomerInquiry> SubEntities
) : IEntity<CustomerInquiry>;

/// <summary>
/// Model to create a Customer Support Department, passed in request body of the API
/// </summary>
public record SupportDepartmentInput(
    string Title,
    string Description,
    string TeamChannelId,
    string TeamId,
    string GroupId,
    string TenantId
);
